src/geoserver-create-imagemosaic-datastore/index.js

```diff
@@ -0,0 +1,138 @@
+import { GeoServerRestClient } from 'geoserver-node-client';
+import { log, initialize } from '../workerTemplate.js';
+import fsPromises from 'fs/promises';
+import { exec } from 'child_process';
+
+const url = process.env.GEOSERVER_REST_URL;
+const user = process.env.GEOSERVER_USER;
+const pw = process.env.GEOSERVER_PASSWORD;
+const workerQueue = process.env.WORKERQUEUE;
+const resultQueue = process.env.RESULTSQUEUE;
+const rabbitHost = process.env.RABBITHOST;
+const rabbitUser = process.env.RABBITUSER;
+const rabbitPass = process.env.RABBITPASS;
+const pgPassword = process.env.POSTGRES_PASSWORD;
+
+const grc = new GeoServerRestClient(url, user, pw);
+
+/**
+ * Creates a imagemosaic datastore in GeoServer based on template configuration files
+ *
+ * @param {Object} workerJob The job object
+ * @param {Array} inputs The inputs for this process
+ *   First input is the workspace to publish to
+ *   Second input is the name of the coverage store to be created
+ * @example
+    {
+       "id": 123,
+       "type": "geoserver-create-imagemosaic-datastore",
+       "inputs": [
+           "klips",
+           "my-coverageStore"
+        ]
+    }
+ */
+const geoserverCreateImageMosaicDatastore = async (workerJob, inputs) => {
+  const ws = inputs[0];
+  const covStore = inputs[1];
+  const geoServerAvailable = await isGeoServerAvailable();
+
+  if (!geoServerAvailable) {
+    console.log('Geoserver not available');
+    console.log('Job should be requeued!');
+    workerJob.missingPreconditions = true;
+    return;
+  }
+
+  try {
+    // check if coverage store exists
+    const covStoreObject = await grc.datastores.getCoverageStore(ws, covStore);
+
+    if (covStoreObject) {
+      log("Datastore already exists!")
+      return;
```

Comment:
Idea: could return sth. like
```
workerJob.status = 'success';
workerJob.outputs = [covStoreObject.name];
```
so that we handle this liek a real success?